File: ConcatenationOfArray1929.java
public class ConcatenationOfArray1929 {
    public int[] getConcatenation(int[] nums) {
        int[] numTotal = new int[nums.length * 2];
        for (int i = 0; i < nums.length; i++) {
            numTotal[i] = nums[i];
            numTotal[i + nums.length] = nums[i];
        }
        return numTotal;
    }

    public static void main(String[] args) {
        ConcatenationOfArray1929 sol = new ConcatenationOfArray1929();

        int[] nums1 = { 1, 2, 1 };
        int[] result1 = sol.getConcatenation(nums1);
        System.out.print("Caso: ");
        for (int num : result1) {
            System.out.print(num + " ");
        }
    }
}

File: Exercicios-JavaScript/ConcatenationOfArray1929.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var getConcatenation = function(nums) {
    let n = nums.length;
    for(let i=0;i<n;i++){
        nums[i + n] = nums[i];
    }
    return nums;
};

console.log(getConcatenation([1,2,1]));